packages/terra-clinical-data-grid/src/DataGrid.jsx

```diff
@@ -0,0 +1,1200 @@
+   */
+  onRowSelect: PropTypes.func,
+  /**
+   * Boolean indicating whether or not resizable columns are enabled for the DataGrid. If this prop is not enabled, the isResizable value of columns
+   * will be ignored.
+   */
+  hasResizableColumns: PropTypes.bool,
+  /**
+   * Number indicating the default column width in px. This value will be used if no overriding width value is provided on a per-column basis.
+   */
+  defaultColumnWidth: PropTypes.number,
+  /**
+   * Function that will be called when the DataGrid's vertical overflow reaches its terminal position. This can be used to contextually
+   * load additional content in the DataGrid. If there is no additional content to present, this function should not be provided.
+   * The `fill` prop must also be provided as true; otherwise, the DataGrid will not overflow internally and will not know to request more content.
+   * Parameters: `onRequestContent()`
+   */
+  onRequestContent: PropTypes.func,
+  /**
+   * Boolean that indicates whether or not the DataGrid should fill its parent container.
+   */
+  fill: PropTypes.bool,
+};
+
+const defaultProps = {
+  pinnedColumns: [],
+  overflowColumns: [],
+  rowHeight: '2.5rem',
+  headerHeight: '3rem',
+  defaultColumnWidth: 200,
+  sections: [],
+};
+
+/**
+ * The VOID_COLUMN_WIDTH value controls the trailing empty column size. This empty column is used
+ * as negative space to allow for resizing of the last overflow column.
+ */
+const VOID_COLUMN_WIDTH = 150;
+
+/**
+ * The PAGED_CONTENT_OFFSET_BUFFER represents the vertical space (in px) remaining in the vertical overflow
+ * that will trigger additonal content retrieval (if onRequestContent is provided).
+ */
+const PAGED_CONTENT_OFFSET_BUFFER = 100;
+
+/**
+ * The width of the column used for row selection.
+ */
+const ROW_SELECTION_COLUMN_WIDTH = 50;
+
+/* eslint-disable react/sort-comp */
+class DataGrid extends React.Component {
+  /**
+   * Returns an Array of HTMLElements that are children of the given 'element' parameter and have
+   * the data-accessible-data-grid-content attribute defined.
+   * @param {HTMLElement} element The element within which to search for accessible content.
+   */
+  static getAccessibleContents(element) {
+    const accessibleArray = [];
+    const accessibleContentNodes = element.querySelectorAll('[data-accessible-data-grid-content]');
+
+    for (let i = 0, numberOfNodes = accessibleContentNodes.length; i < numberOfNodes; i += 1) {
+      accessibleArray.push(accessibleContentNodes[i]);
+    }
+
+    return accessibleArray;
+  }
+
+  /**
+   * Returns the configuration for the row selection column.
+   */
+  static getRowSelectionColumn() {
+    return {
+      id: 'DataGrid-rowSelectionColumn',
+      width: ROW_SELECTION_COLUMN_WIDTH,
+    };
+  }
+
+  /**
+   * Returns the configuration for the void column utilized for column resizing.
+   */
+  static getVoidColumn() {
```

Comment:
Would it be simpler to define getRowSelectionColumn and getVoidColumn as objects instead of functions? This would match https://github.com/cerner/terra-clinical/pull/387/files#diff-93badd98f0bc3febc86ceb7ac7123ad8R110.

Reply:
I do think that would make more sense.